ChatQuickstart/QuickstartChatManager.swift

```diff
@@ -0,0 +1,141 @@
+                                        self.generalChannel?.setUniqueName(defaultChannel, completion: { result in
+                                            print("channel unique name set")
+                                        })
+                                    })
+                                }
+                        })
+                    }
+                })
+            }
+        }
+    }
+    
+    // Called whenever a channel we've joined receives a new message
+    func chatClient(_ client: TwilioChatClient, channel: TCHChannel,
+                    messageAdded message: TCHMessage) {
+        self.messages.append(message)
+
+        DispatchQueue.main.async() {
+            if let delegate = self.chatManagerDelegate {
+                delegate.reloadMessages()
+                if self.messages.count > 0 {
+                    delegate.scrollToBottomMessage()
+                }
+            }
+        }
+    }
+    
+    func getMessages() -> [TCHMessage] {
+        return messages
+    }
+    
+    func sendMessage(_ messageText:String, completion: @escaping (TCHResult, TCHMessage?) -> Void) {
+        if let messages = self.generalChannel?.messages {
+            let messageOptions = TCHMessageOptions().withBody(messageText)
+            messages.sendMessage(with: messageOptions, completion: { (result, message) in
+                completion(result, message)
+            })
+        }
+    }
+    
+    
+    func login(_ identity:String, completion:@escaping (TCHResult)->Void) {
+        // Fetch Access Token from the server and initialize Chat Client - this assumes you are
+        // calling a Twilio function, as described in the Quickstart docs
+        let deviceId = UIDevice.current.identifierForVendor!.uuidString
+        let urlString = "\(tokenURL)?identity=\(identity)&device=\(deviceId)"
+        
+        TokenUtils.retrieveToken(url: urlString) { (token, identity, error) in
+            if let token = token {
+                // Set up Twilio Chat client
+                TwilioChatClient.chatClient(withToken: token, properties: nil, delegate: self) {
+                    (result, chatClient) in
+                        self.client = chatClient;
+                        self.createGeneralChannelIfNeeded()
+                        completion(result)
+                    }
+            } else {
+                print("Error retrieving token: \(error.debugDescription)")
+            }
+            
+        }
+    }
+    
+    
+    func logout() {
+        if let client = client {
+            client.delegate = nil
+            client.shutdown()
+            self.client = nil
+        }
+    }
+    
+    func createGeneralChannelIfNeeded() {
+        // Create the general channel (for public use) if it hasn't been created yet
+        
+        let options = [
+            TCHChannelOptionFriendlyName: "General Channel",
+            TCHChannelOptionType: TCHChannelType.public.rawValue
```

Comment:
With the general shift towards Conversations, which only have PRIVATE channels I think it would make sense to put more accent on creating and using PRIVATE channels as opposed to PUBLIC ones.